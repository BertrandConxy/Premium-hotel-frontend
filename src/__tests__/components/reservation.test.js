import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Reservation from '../../components/Reservation';
import store from '../../redux/configureStore';

const reservation = [
  {
    check_in: '2000-01-01',
    check_out: '2000-01-02',
    room_id: 1,
    id: 1,
  },
];
const ReservationProvider = () => (
  <Provider store={store}>
    <Reservation reservation={reservation} />
  </Provider>
);

describe('render component', () => {
  it('renders the reservation details', () => {
    render(<ReservationProvider />);
    const checkIn = screen.getByText('Check-in:');
    const checkOut = screen.getByText('Check-out:');
    expect(checkIn).toBeInTheDocument();
    expect(checkOut).toBeInTheDocument();
  });
});

describe('Snapshot test', () => {
  it('looks the same as the snapshot', () => {
    const component = renderer.create(<ReservationProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
