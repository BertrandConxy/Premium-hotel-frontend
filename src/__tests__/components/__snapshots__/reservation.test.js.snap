// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<div
  className="reservation"
>
  <div
    className="co1"
  >
    <div
      className="check-in tex"
    >
      <h3>
        Check-in:
      </h3>
      <span />
    </div>
    <div
      className="check-out tex"
    >
      <h3>
        Check-out:
      </h3>
      <span />
    </div>
    <div
      className="Nights tex"
    >
      <h3>
        Nights:
      </h3>
      <span>
        NaN
      </span>
    </div>
  </div>
  <div
    className="co2"
  >
    <div
      className="room-id tex"
    >
      <h3>
        Room-id:
      </h3>
      <span>
        â„–
      </span>
    </div>
    <div
      className="rate-id tex"
    >
      <h3>
        Total Rate:
      </h3>
      <span>
        $
        NaN
      </span>
    </div>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
